core/dbt/adapters/base/connections.py

```diff
@@ -142,44 +142,44 @@ def exception_handler(self, sql: str) -> ContextManager:
         )
 
     def set_connection_name(self, name: Optional[str] = None) -> Connection:
-        conn_name: str
-        if name is None:
-            # if a name isn't specified, we'll re-use a single handle
-            # named 'master'
-            conn_name = "master"
-        else:
-            if not isinstance(name, str):
-                raise dbt.exceptions.CompilerException(
-                    f"For connection name, got {name} - not a string!"
-                )
-            assert isinstance(name, str)
-            conn_name = name
+        """Called by 'acquire_connection' in BaseAdapter, which is called by
+        'connection_named', called by 'connection_for(node)'.
+        Creates a connection for this thread if one doesn't already
+        exist, and will rename an existing connection."""
 
+        conn_name: str = "master" if name is None else name
+
+        # Get a connection for this thread
         conn = self.get_if_exists()
+
+        if conn and conn.name == conn_name and conn.state == "open":
```

Comment:
Do we only consider a connection to be reused if the name changed? Reading the code I would expect this to also be a connection reused.

Reply:
It's currently not considered reused. Of course the code is so strangely factored that it's a bit hard to tell how it ought to be used.